components/cmp/services/src/CmpWaitForLibrary/index.js

```diff
@@ -0,0 +1,49 @@
+import {Component} from 'react'
+import PropTypes from 'prop-types'
+
+export class CmpWaitForLibrary extends Component {
+  state = {cmpReady: this.props.cmpReady}
+
+  componentDidMount() {
+    // check if the cmp is already ready in order to
+    // avoid adding listeneres and check its availability
+    if (this.props.cmpReady === true) return
```

Comment:
I don't think you need this `cmpReady`. CmpWaitForLibrary could control it seamlessly.
Once it's loaded, it's loaded forever, right ?
You can memoize it here, I think.

Reply:
Mmmm I'm doing this as the Modal and the Banner are using this component. In both I can to check if the cmp is ready but there's a case when I don't want to check. It's when the Modal is used inside the Banner. In that case, I'm using this prop to bypass the check in the Modal as is unnecesary.

I mean, I could create a `let` outside the class in order to do the `memoization` but I don't know if I would be happy with it.

Reply:
Why ? I don't see the problem. The value would be to avoid unnecessary props but I don't see the value of avoiding let.
Anyways as it's only used internally It's not that important.